examples/telco/notebook.jl

```diff
@@ -0,0 +1,756 @@
+# | `auc(yhat, y)`, `brier_loss(yhat, y)` | evaluate two probabilistic measures (`yhat` a vector of probability distributions)
+# | `machine(model, X, y)` | bind `model` to training data `X` (features) and `y` (target)
+# | `fit!(mach, rows=...)` | train machine using specified rows (observation indices)
+# | `predict(mach, rows=...)`, | make in-sample model predictions given specified rows
+# | `predict(mach, Xnew)` | make predictions given new features `Xnew`
+# | `fitted_params(mach)` | inspect learned parameters
+# | `report(mach)`        | inspect other outcomes of training
+# | `confmat(yhat, y)`    | confusion matrix for predictions `yhat` and ground truth `y`
+# | `roc(yhat, y)` | compute points on the receiver-operator Characteristic
+# | `StratifiedCV(nfolds=6)` | 6-fold stratified cross-validation resampling strategy
+# | `Holdout(fraction_train=0.7)` | holdout resampling strategy
+# | `evaluate(model, X, y; resampling=..., options...)` | estimate performance metrics `model` using the data `X`, `y`
+# | `FeatureSelector()` | transformer for selecting features
+# | `Step(3)` | iteration control for stepping 3 iterations
+# | `NumberSinceBest(6)`, `TimeLimit(60/5), InvalidValue()` | iteration control stopping criteria
+# | `IteratedModel(model=..., controls=..., options...)` | wrap an iterative `model` in control strategies
+# | `range(model,  :some_hyperparam, lower=..., upper=...)` | define a numeric range
+# | `RandomSearch()` | random search tuning strategy
+# | `TunedModel(model=..., tuning=..., options...)` | wrap the supervised `model` in specified `tuning` strategy
+
+
+# ## Instantiate a Julia environment
+
+# The following code replicates precisely the set of Julia packages
+# used to develop this tutorial. If this is your first time running
+# the notebook, package instantiation and pre-compilation may take a
+# minute or so to complete. **This step will fail** if the [correct
+# Manifest.toml and Project.toml
+# files](https://github.com/alan-turing-institute/MLJ.jl/tree/dev/examples/telco)
+# are not in the same directory as this notebook.
+
+using Pkg
+Pkg.activate(@__DIR__) # get env from TOML files in same directory as this notebook
+Pkg.instantiate()
+
+
+# ## Warm up: Building a model for the iris dataset
+
+# Before turning to the Telco Customer Churn dataset, we very quickly
+# build a predictive model for Fisher's well-known iris data set, as way of
+# introducing the main actors in any MLJ workflow. Details that you
+# don't fully grasp should become clearer in the Telco study.
+
+# This section is a condensed adaption of the [Getting Started
+# example](https://alan-turing-institute.github.io/MLJ.jl/dev/getting_started/#Fit-and-predict)
+# in the MLJ documentation.
+
+# First, using the built-in iris dataset, we load and inspect the features
+# `X_iris` (a table) and target variable `y_iris` (a vector):
+
+using MLJ
+
+#-
+
+const X_iris, y_iris = @load_iris;
```

Comment:
it is not clear if making a `const` is needed for performance or not.

Reply:
Yeah, I guess not. So you think it's clearer to just drop it?

Reply:
I just wanted to make sure. Normally when one sees `const` it signals that this is needed for performance (the reason is that `const` does not preclude changing of the value bound to the variable name - it only fixes the type).

Reply:
Mmm, just tested this, and it did not work for me. I'm guessing that  activating an environment in your startup.jl overrides this behaviour (and probably lost of users do that). Seems to me the safest thing is to leave the explicit specification of environment as is, unless you have a better suggestion.

Reply:
OK